src/Engine/KeyMap/KeyMap.cpp

```diff
@@ -0,0 +1,121 @@
+#include "KeyMap.h"
+#include "Engine/Input/InputChecker.h"
+
+#include<fstream>
+#include<string>
+#include<vector>
+#include<iostream>
+
+std::string path = "editor.binds";
+
+std::ostream& operator << (std::ostream& os, const EditorAction& obj)
+{
+   os << static_cast<std::underlying_type<EditorAction>::type>(obj);
+   return os;
+}
+
+std::ostream& operator << (std::ostream& os, const InputType& obj)
+{
+   os << static_cast<std::underlying_type<InputType>::type>(obj);
+   return os;
+}
+
+Bind KeyMap::CreateBind(EditorAction action, InputType input_type, std::vector<SDL_KeyCode> keys) {
+    Bind bind = {action, input_type, keys};
+    return bind;
+}
+
+
+KeyMap::KeyMap() {
+    m_Default_Bindings_Map = { 
+        {EditorAction::EXIT_CURRENT_TOOL, CreateBind(EditorAction::ENTER_DRAW_TOOL, InputType::LATCH, std::vector<SDL_KeyCode>{SDLK_ESCAPE})},
```

Comment:
Should EXIT_CURRENT_TOOL be mapped to ENTER_DRAW_TOOL?

Reply:
No. Whoops. I'll get that fixed.